### PR TITLE
feat: migrate grok to new parser style

### PR DESCRIPTION
**Pull request by powersj**
fixes: #11357

### CODE REVIEW

**Comment by powersj** [comment]
@srebhan I will have a couple of test failures on this one:

1. Couple in config, complaining about `Difference in SetParser() for "grok"` where should I look for that?
2. `inputs/logparser` had a test that expects to get an error with no parser set. Should I change how I build the parser? or add some sort of check?

File: plugins/inputs/logparser/logparser.go [hunk 1]
```diff
@@ -128,11 +129,12 @@ func (l *LogParserPlugin) Start(acc telegraf.Accumulator) error {
 		DataFormat:             "grok",
 	}
 
-	var err error
-	l.GrokParser, err = parsers.NewParser(config)
+	parser := grok.Parser{}
+	err := parser.InitFromConfig(config)
```
Comment (srebhan) [inline plugins/inputs/logparser/logparser.go:133]: Can you please initialize the parser directly instead of using the old, to-be-removed, interface!?

**Review by srebhan — APPROVED** [review]
The code looks good. Thanks @powersj!
Regarding the config-test-fail, you probably need [this fix](https://github.com/influxdata/telegraf/pull/11405/commits/4cf589da603c37906f85445aefec4212342cd008) to ensure the parser is initialized. See my comment in #11407.

**Comment by srebhan** [comment]
I do agree with your assessment.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @powersj!

**Pull request merged**